pimd/pim_upstream.c

```diff
@@ -1830,30 +1830,31 @@ bool pim_upstream_equal(const void *arg1, const void *arg2)
  */
 static bool pim_upstream_kat_start_ok(struct pim_upstream *up)
 {
-	struct pim_instance *pim = up->channel_oil->pim;
+	struct channel_oil *c_oil = up->channel_oil;
+	struct interface *ifp = up->rpf.source_nexthop.interface;
+	struct pim_interface *pim_ifp;
+
+	/* "iif == RPF_interface(S)" check is not easy to do as the info
+	 * we get from the kernel/ASIC is really a "lookup/key hit".
+	 * So we will do an approximate check here to avoid starting KAT
+	 * because of (S,G,rpt) forwarding on a non-LHR.
+	 */
+	if (!ifp)
+		return false;
+
+	pim_ifp = ifp->info;
+	if (pim_ifp->mroute_vif_index != c_oil->oil.mfcc_parent)
+		return false;
 
-	/* "iif == RPF_interface(S)" check has to be done by the kernel or hw
-	 * so we will skip that here */
 	if (up->rpf.source_nexthop.interface &&
```

Comment:
We can remove the first condition here as we have done that already :)

Reply:
ack. will fixup.

Reply:
I have updated the PR with a new commit that does the fixup.